crates/sui-types/src/transaction.rs

```diff
@@ -947,6 +949,31 @@ impl ProgrammableTransaction {
             command.validity_check(config)?;
         }
 
+        // A command that uses Random can only be followed by TransferObjects or MergeCoins.
+        if config.enable_randomness_ptb_limits() {
+            // Check if there is a random object in the input objects
+            if let Some(random_index) = input_objects.iter().position(|obj| {
```

Comment:
You will need all of the `inputs` not just `input_objects`. So 
```suggestion
            if let Some(random_index) = inputs.iter().position(|obj| {
```
with then fixing the predicate